processor/samplingprocessor/tailsamplingprocessor/sampling/numeric_tag_filter.go

```diff
@@ -44,31 +43,53 @@ func NewNumericAttributeFilter(logger *zap.Logger, key string, minValue, maxValu
 // after the sampling decision was already taken for the trace.
 // This gives the evaluator a chance to log any message/metrics and/or update any
 // related internal state.
-func (naf *numericAttributeFilter) OnLateArrivingSpans(Decision, []*tracepb.Span) error {
+func (naf *numericAttributeFilter) OnLateArrivingSpans(Decision, []*pdata.Span) error {
 	naf.logger.Debug("Triggering action for late arriving spans in numeric-attribute filter")
 	return nil
 }
 
 // Evaluate looks at the trace data and returns a corresponding SamplingDecision.
 func (naf *numericAttributeFilter) Evaluate(_ pdata.TraceID, trace *TraceData) (Decision, error) {
-	naf.logger.Debug("Evaluating spans in numeric-attribute filter")
 	trace.Lock()
 	batches := trace.ReceivedBatches
 	trace.Unlock()
 	for _, batch := range batches {
-		for _, span := range batch.Spans {
-			if span == nil || span.Attributes == nil {
+		rspans := batch.ResourceSpans()
+		for i := 0; i < rspans.Len(); i++ {
+			rs := rspans.At(i)
+			if rs.IsNil() {
 				continue
 			}
-			if v, ok := span.Attributes.AttributeMap[naf.key]; ok {
-				value := v.GetIntValue()
-				if value >= naf.minValue && value <= naf.maxValue {
-					return Sampled, nil
+			resource := rs.Resource()
+			if !resource.IsNil() {
+				if v, ok := resource.Attributes().Get(naf.key); ok {
+					value := v.IntVal()
+					if value >= naf.minValue && value <= naf.maxValue {
+						return Sampled, nil
```

Comment:
Isn't this changing the behavior of the filter? Previously, it was acting only on span attributes, not resources, no?

If we really want to change the behavior in this PR, we should then at least extract the filter logic into its own function, receiving a list of attributes. The same logic can then be applied to resource attributes and to span attributes.

Reply:
Removed : i will create a new PR with the resource attribute filter. For now removed this functionality and its test case from this PR. Thanks